### PR TITLE
add some docstrings

### PR DESCRIPTION
**Pull request by bw55555**

### CODE REVIEW

File: src/main/java/interfaces/State.java [hunk 1]
```diff
@@ -1,8 +1,24 @@
...
     public boolean postInput(String input);
+
+    /**
+     * @return whether the state is awaiting input
+     */
     public boolean awaitInput();
+
+    /**
+     * @return the input validator for accepted inputs
+     */
```
Comment (mahirhassan-uoft) [inline src/main/java/interfaces/State.java:22]: doctest info is clear

**Review by mahirhassan-uoft — APPROVED** [review]
minor changes, but the overall input validator and states look structurally correct
solid work 👍

**Review by marjaleenakiik — APPROVED** [review]
Docstrings are descriptive while not being too long.

**Comment by Yasutakahi** [comment]
It looks very nice

**Comment by juhwansonn** [comment]
looks very precise and descriptive

**Comment by vegetablezoe** [comment]
Looking good

**Comment by passionfruitjello** [comment]
docstrings looking good

**Review by Yasutakahi — APPROVED** [review]
It is easy to understand and clear, so I think it is nice!!!

**Pull request merged**